### ISSUE TITLE
Trace ranges greater than a single function with Intel PT

### ISSUE THREAD

**Issue by MatthewTingum** [googleprojectzero/winafl#320]
I'll preface by saying that my knowledge of AFL and Intel PT is limited at the time being.  This issue is  in regards to Intel PT instrumentation. Maybe it is better to say that it pertains to Intel PT instrumentation. Digging deeper on the term "instrumentation", maybe that is not the right word. I digress.

Projects like kAFL support tracing of large ranges. In fact, their documentation suggests tracing entire drivers. I understand that this project is *not* kAFL and winafl faces different limitations. That said, it seems like Intel PT could make use of arbitrary ranges. Does it do that already? It's not clear in the documentation. Of course, if it does not support that, I wouldn't expect to find that in the documentation.

My questions are:
- Is there a reasonable limit to the range Intel PT can trace?
  - I'm thinking there is. Can a user specify some limit?
- Was PT retrofitted DynamoRIO APIs?

In short, I'm proposing arbitrary trace ranges with Intel PT.

**Comment by ifratric** [comment]
Hey, there are a couple of things to consider here. In the subject line implies that WinAFL traces only a "single function", but WinAFL's Intel PT mode does not trace a single function, in fact it traces the entire process. By "single function", I assume you mean `-target_method`, but that's actually related to persistent mode fuzzing and not instrumentation itself. When the trace data (for the entire process) is being processed, the coverage data is computed only for modules specified using `-coverage_module` flag (multiple modules can be specified this way).

The second thing to consider here is specifying IP ranges to traces in order to *limit* trace data. This is something Intel PT itself supports, but at the time PT mode in WinAFL was written, this feature wasn't exposed by the PT Windows driver. This might have changed in the meantime.
